src/gc/gc.d

```diff
@@ -1969,7 +1954,11 @@ struct Gcx
             enum initSize = 64 * 1024; // Windows VirtualAlloc granularity
             immutable ncap = _cap ? 2 * _cap : initSize / Range.sizeof;
             auto p = cast(Range*)os_mem_map(ncap * Range.sizeof);
-            if (p is null) onOutOfMemoryErrorNoGC();
+            if (p is null)
+            {
+                GC.gcLock.unlock();
```

Comment:
same here - this style of pairing lock/unlock across different scopes is dangerous